### ISSUE TITLE
💅 noUselessUndefined removes useful undefined in return

### ISSUE THREAD

**Issue by hyperknot** [biomejs/biome#7804]
### Environment information

```bash
playground
```

### Rule name

lint/nursery/noUselessUndefined

### Playground link

https://next.biomejs.dev/playground/?lintRules=noUselessUndefined&code=IAAgAGMAbwBuAHMAdAAgAGMAbwBtAG0AbwBuAEMAbwBuAGYAaQBnACAAPQAgAHsACgAgACAAIAAgAHQAbwBrAGUAbgBpAHoAZQByADoAIAB7AAoAIAAgACAAIAAgACAAZABlAGwAKABfAHMAcgBjADoAIABzAHQAcgBpAG4AZwApACAAewAKACAAIAAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgAHUAbgBkAGUAZgBpAG4AZQBkAAoAIAAgACAAIAAgACAAfQAsAAoAIAAgACAAIAB9ACwACgAgACAAfQAKAAoAIAAgAHAAcgBvAGMAZQBzAHMAbwByAFMAaQBtAHAAbABlACAAPQAgAG4AZQB3ACAATQBhAHIAawBlAGQAKABjAG8AbQBtAG8AbgBDAG8AbgBmAGkAZwApAA%3D%3D&ruleDomains.solid=recommended

### Expected result

What can I do, if TypeScript requires undefined in the return type, and noUselessUndefined is trying to remove it? My best idea is to say `return undefined as undefined`.

Otherwise I get `Type void is not assignable to type false | Del | undefined` in TS error.

I mean `return` and `return undefined` are different return types in TS. Maybe return shouldn't be cleaned up then?

Also, now that Biome is doing some kind of type checks, do you think you could go as far as to only remove undefined if the type allows it?

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It seems like the rule is working as expected, and TypeScript accepts the conditions without errors https://www.typescriptlang.org/play/?#code/ATDGHsDsGcBc3AW0VAwlAZgSwObALzADeAUCCLOANYCmkWAXjQE4BcxZ5IAJjQDYAKAPrRmodnGZZIOAJTsArpF7ZINbhy5dmNWAuaQA3Jy4BfADQngFzqZKcADs3Cga0aOGYBlLIgd8aAmA1AHdgAFkAQ2ZabgEIZDRMXFkgA
